dbms/src/Storages/MergeTree/MergeTreeDataSelectExecutor.cpp

```diff
@@ -1013,20 +1043,31 @@ BlockInputStreams MergeTreeDataSelectExecutor::spreadMarkRangesAmongStreamsWithO
                     sorting_info->direction, 1);
 
             for (auto & stream : streams_per_thread)
-                stream = std::make_shared<ExpressionBlockInputStream>(stream, sorting_key_prefix_expr);
+            {
+                pipe.emplace_back(std::make_shared<ExpressionTransform>(stream->getHeader(), sorting_key_prefix_expr));
+                connect(*stream, pipe.back()->getInputs().front());
```

Comment:
Connection from `stream` to `pipe.back()->getInputs().front()`?  I thought our graph is directed.

Reply:
That's correct. Graph is directed. We do connect output port and input port.